### PR TITLE
feat: remove pair

### PR DESCRIPTION
**Pull request by janfabian**
## Summary of changes

implemnted `ExecuteMsg::RemovePair` as described in https://github.com/White-Whale-Defi-Platform/migaloo-core/issues/38

## Report of required housekeeping

- [x] Github issue OR spec proposal link https://github.com/White-Whale-Defi-Platform/migaloo-core/issues/38
- [x] Wrote tests
- [ ] Added a relevant changelog entry to CHANGELOG.md

----

## (FOR ADMIN) Before merging

- [ ] Added appropriate labels to PR
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" (coding standards)
- [ ] Confirm added tests are consistent with the intended behavior of changes
- [ ] Ensure all tests pass

### CODE REVIEW

File: terraswap_factory/src/testing.rs [hunk 1]
```diff
@@ -941,3 +941,98 @@ fn failed_migrate_pair_with_no_admin() {
...
+        pair_address: String::from_utf8(pair_key_vec.clone()).unwrap(),
+    };
+    let env = mock_env();
+    let info = mock_info("addr0000", &[]);
+    let res = execute(deps.as_mut(), env, info, msg);
+
+    assert!(res.is_err(), "Response should be error");
+    assert_eq!(
+        res.err().unwrap().to_string(),
+        "Generic error: Pair doesnt exists"
```
Comment (kaimen-sano) [inline terraswap_factory/src/testing.rs:1036]:
```suggestion
        res.unwrap_err(),
        StdError::generic_err("Pair doesn't exist")
```

File: terraswap_factory/src/testing.rs [hunk 2]
```diff
@@ -941,3 +941,98 @@ fn failed_migrate_pair_with_no_admin() {
...
+    let pair_key_vec = pair_key(&raw_infos);
+
+    let msg = ExecuteMsg::RemovePair {
+        pair_address: String::from_utf8(pair_key_vec.clone()).unwrap(),
+    };
+    let env = mock_env();
+    let info = mock_info("addr0000", &[]);
+    let res = execute(deps.as_mut(), env, info, msg);
+
+    assert!(res.is_err(), "Response should be error");
```
Comment (kaimen-sano) [inline terraswap_factory/src/testing.rs:1033]: Should be able to remove this line with the below change

File: terraswap_factory/src/commands.rs [hunk 3]
```diff
@@ -121,6 +121,17 @@ pub fn create_pair(
         }))
 }
 
+pub fn remove_pair(deps: DepsMut, _env: Env, pair_key: String) -> StdResult<Response> {
+    let pair_key_bytes = pair_key.as_bytes();
+    if let Ok(None) = PAIRS.may_load(deps.storage, pair_key_bytes) {
+        return Err(StdError::generic_err("Pair doesnt exists"));
```
Comment (kaimen-sano) [inline terraswap_factory/src/commands.rs:127]: Slight typo here, should be "Pair doesn't exist"

**Review by kaimen-sano — CHANGES_REQUESTED** [review]
Nice small changes, just some slight improvements to the tests here!

File: terraswap_factory/src/commands.rs [hunk 4]
```diff
@@ -121,6 +121,17 @@ pub fn create_pair(
         }))
 }
 
+pub fn remove_pair(deps: DepsMut, _env: Env, pair_key: String) -> StdResult<Response> {
```
Comment (kerber0x) [inline terraswap_factory/src/commands.rs:124]: `pair_key` should be `pair_address` in the function for consistency

File: terraswap_factory/src/commands.rs [hunk 5]
```diff
@@ -121,6 +121,17 @@ pub fn create_pair(
...
+    let pair_key_bytes = pair_key.as_bytes();
+    if let Ok(None) = PAIRS.may_load(deps.storage, pair_key_bytes) {
+        return Err(StdError::generic_err("Pair doesnt exists"));
+    }
+
+    PAIRS.remove(deps.storage, pair_key_bytes);
+
+    return Ok(Response::new().add_attributes(vec![("pair_contract_addr", pair_key)]));
```
Comment (kerber0x) [inline terraswap_factory/src/commands.rs:132]: Please add what action is being executed in the attributes

**Review by kerber0x — CHANGES_REQUESTED** [review]
Great job! I only have some minor suggestions, please take a look

**Comment by janfabian** [comment]
@kaimen-sano @kerber0x thanks guys for the feedback, it should be solved 

I also open PR to `core` with the RemovePair message https://github.com/White-Whale-Defi-Platform/migaloo-core/pull/87. I'm not sure how to maintain the order of the PRs

**Review by kerber0x — APPROVED** [review]
Good job!

**Comment by kerber0x** [comment]
> @kaimen-sano @kerber0x thanks guys for the feedback, it should be solved
> 
> I also open PR to `core` with the RemovePair message [White-Whale-Defi-Platform/migaloo-core#87](https://github.com/White-Whale-Defi-Platform/migaloo-core/pull/87). I'm not sure how to maintain the order of the PRs

Awesome

> @kaimen-sano @kerber0x thanks guys for the feedback, it should be solved
> 
> I also open PR to `core` with the RemovePair message [White-Whale-Defi-Platform/migaloo-core#87](https://github.com/White-Whale-Defi-Platform/migaloo-core/pull/87). I'm not sure how to maintain the order of the PRs

That's awesome, thanks for your contribution! we decided to merge the pool-network repo into migaloo-core, so we won't need to deal with submodules :) Don't worry about the PR on the core repo, we will deal with it. Thanks!

**Pull request closed without merging**